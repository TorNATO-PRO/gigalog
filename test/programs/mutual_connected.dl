link(a, b).
link(b, c).
link(c, d).
link(d, e).

path(X, Y) :- link(X, Y).
path(X, Y) :- step(X, Y).

step(X, Y) :- link(X, Z), path(Z, Y).
