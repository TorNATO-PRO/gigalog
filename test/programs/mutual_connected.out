link(a,b)
link(b,c)
link(c,d)
link(d,e)
path(a,b)
path(a,c)
path(a,d)
path(a,e)
path(b,c)
path(b,d)
path(b,e)
path(c,d)
path(c,e)
path(d,e)
step(a,c)
step(a,d)
step(a,e)
step(b,d)
step(b,e)
step(c,e)
